### PR TITLE
Cache Class for Performance Counters

### PR DESCRIPTION
**Pull request by frackleton**
continued from PR into master, changed target branch

### CODE REVIEW

File: AggregateMetrics/AggregateMetrics/AzureWebApp/FlexiblePerformanceCounterGauge.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        public FlexiblePerformanceCounterGauge(string name)
+        {
+            this.name = name;
+        }
+            /// <summary>
+            /// Returns the current value of the counter as a <c ref="MetricTelemetry"/> and resets the metric.
+            /// </summary>
+            /// <returns> Metric Telemetry object mt, with values for Name and Value </returns>
+
+            public MetricTelemetry GetValueAndReset()
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/FlexiblePerformanceCounterGauge.cs]: styling again

File: AggregateMetrics/AggregateMetrics/AzureWebApp/CacheHelper.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                /* http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_ASPNET/
+                http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_APP/
+                http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_CLR/
+               http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_ALL/ */
+                Task<string> counterRetrieval = client.GetStringAsync("http://remoteenvironmentvariables.azurewebsites.net/api/EnvironmentVariables/WEBSITE_COUNTERS_APP/");
+                counterRetrieval.Wait();
+
+                string uncachedJson = counterRetrieval.Result;
+
+                CacheHelper.SaveToCache(name, uncachedJson, DateTimeOffset.Now.AddSeconds(5.0));
```
Comment (SergeyKanzhelev) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/CacheHelper.cs]: there were a comment before that you need to cache an entire json

**Comment by SergeyKanzhelev** [comment]
merging for now to unblock Mateo

File: AggregateMetrics/AggregateMetrics/AzureWebApp/CacheHelper.cs (hunk 2 above)
Comment (frackleton) [inline AggregateMetrics/AggregateMetrics/AzureWebApp/CacheHelper.cs]: uncachedJson is the entire json that is retrieved from the HTTP get

**Pull request merged**